stdcommands/setstatus/setstatus.go

```diff
@@ -12,17 +12,40 @@ var Command = &commands.YAGCommand{
 	CmdCategory:          commands.CategoryDebug,
 	HideFromCommandsPage: true,
 	Name:                 "setstatus",
-	Description:          "Sets the bot's status and optional streaming url. Bot Admin Only",
+	Description:          "Sets the bot's presence type, status text, online status, and optional streaming URL. Bot Admin Only",
 	HideFromHelp:         true,
 	Arguments: []*dcmd.ArgDef{
 		{Name: "status", Type: dcmd.String, Default: ""},
 	},
 	ArgSwitches: []*dcmd.ArgDef{
-		{Name: "url", Type: dcmd.String, Default: ""},
+		{Name: "url", Type: dcmd.String, Help: "The URL to the stream. Must be on twitch.tv or youtube.com. Activity type will always be streaming if this is set.", Default: ""},
+		{Name: "type", Type: dcmd.String, Help: "Set activity type. Allowed values are 'playing', 'streaming', 'listening', 'watching', 'custom', 'competing'. Defaults to custom status", Default: "custom"},
+		{Name: "status", Type: dcmd.String, Help: "Set online status. Allowed values are 'online', 'idle', 'dnd', 'offline'. Defaults to online", Default: "online"},
 	},
 	RunFunc: util.RequireBotAdmin(func(data *dcmd.Data) (interface{}, error) {
-		streamingURL := data.Switch("url").Str()
-		bot.SetStatus(streamingURL, data.Args[0].Str())
+		activityType := data.Switch("type").Str()
+		statusType := data.Switch("status").Str()
+		statusText := data.Args[0].Str()
+		streamingUrl := data.Switch("url").Str()
+		switch activityType {
+		case "playing":
+		case "streaming":
+		case "listening":
+		case "watching":
+		case "custom":
+		case "competing":
+		default:
+			activityType = "custom"
```

Comment:
Why not issue an error for unknown activities to catch typos such as `wtching` instead of silently defaulting to `custom`? Users can still set custom status via passing `custom` directly.

Reply:
Uh...not sure of the best way to error 😅 can reference other commands though